extensions/src/org/pentaho/platform/osgi/KarafInstance.java

```diff
@@ -25,29 +48,60 @@
   private String cachePath;
   private FileLock fileLock;
   private String root;
+  private HashMap<String, KarafInstanceProperty> instanceProperties = new HashMap<String, KarafInstanceProperty>();
   private static final int MAX_NUMBER_OF_KARAF_INSTANCES = 50;
   private static final int BANNER_WIDTH = 79;
+  private final PortAssigner portAssigner = new PortAssigner();
+  private boolean started;
+  private static KarafInstance instance; // Not final because unit test creates multiple instances
+
   private StringBuilder banner = new StringBuilder();
 
-  public KarafInstance( String root ) {
+  protected KarafInstance( String root ) {
```

Comment:
Just leave it package protected, not "protected" so that Unit Tests can call the constructor. Everyone else should be calling getInstance() (KarafBoot seems to be calling the constructor)